katcp/kattypes.py

```diff
@@ -923,6 +930,31 @@ def make_reply(msgname, types, arguments, major):
             msgname, *pack_types((Str(),) + types, arguments, major))
     raise ValueError("First returned value must be 'ok' or 'fail'.")
 
+def concurrent_reply(handler):
+    """Decorator for concurrent async request handlers
+
+    By default async request handlers that return a Future are serialised
+    per-connection, i.e. until the most recent handler resolves its future, the
+    next message will not be read from the client stream. A handler decorated
+    with this decorator allows the next message to be read before it has
+    resolved its future, allowing multiple requests from a single client to be
+    handled concurrently. This is similar to raising AsyncReply.
```

Comment:
If I am not mistaken, then this would be a clean katcp way to replace what we hacked for `activate_subarray` in https://github.com/ska-sa/katobs/pull/211/files#diff-6ed26d613abbf9f5bd7b5d6de341109fR1462?